### PR TITLE
refactor: adjust swiper options and add progress component 

### PR DESCRIPTION
**Pull request by doohanas**
### Related
[S8A1 - Membuat Tampilan Widget Slideshow](https://app.clickup.com/t/9003239225/CES-2906)

### Changes
- adjust swiper options 
- add progress component 

### Evidence
title: refactor: adjust swiper options and add progress component 
project: Portal Jabar
participants: @doohanas @azhargraha @marsellavaleria19 @yoslie

### CODE REVIEW

**Review by azhargraha — APPROVED** [review]

File: components/Slideshow.vue [hunk 1]
```diff
@@ -104,12 +106,17 @@
...
   import 'swiper/css/effect-coverflow'
+  import 'swiper/css/autoplay'
 
   /**
    * @TODO: Remove ref when feature caption is ready
    */
   const isHideCaptionFeat = ref(false)
 
+  const initialSlideSwiper = computed(() => {
+    return validatedPayload?.value?.images?.length >= 3 ? 1 : 0
```
Comment (Ibwedagama) [inline components/Slideshow.vue:117]: hello @doohanas validatedPayload is already being validated right? So isn't it somewhat pointless to add optional chaining, no? 🤔

File: components/Slideshow.vue [hunk 2]
```diff
@@ -151,23 +158,60 @@
...
   })
 
+  /**
+   * Start: set swiper options
+   */
+
+  const progressValue = ref(0)
+
+  const isLoop = computed(() => {
+    return validatedPayload?.value?.images?.length > 3 ? true : false
```
Comment (Ibwedagama) [inline components/Slideshow.vue:168]: minor: I think we don't have to explicitly return boolean here @doohanas 

```suggestion
    return validatedPayload?.value?.images?.length > 3
```

File: components/Slideshow.vue [hunk 3]
```diff
@@ -151,23 +158,60 @@
...
    */
 
   const activeItem = ref<string>('01')
 
-  function onSlideChange(params: { activeIndex: number | null }) {
-    if (params.activeIndex !== null) {
-      const activeIndex = params.activeIndex + 1
+  function onSlideChange(params: { realIndex: number }) {
+    console.log(params.realIndex + 1, validatedPayload?.value?.images?.length)
+
```
Comment (Ibwedagama) [inline components/Slideshow.vue:195]: hello @doohanas please remove the console log

File: components/Slideshow.vue [hunk 4]
```diff
@@ -151,23 +158,60 @@
...
+      ? {
+          delay: 5000,
+          disableOnInteraction: false,
+        }
+      : false
+  })
+
+  function getValueProgress(activeIndex: number, maxLength: number) {
+    progressValue.value = (activeIndex / maxLength) * 100
+  }
```
Comment (Ibwedagama) [inline components/Slideshow.vue:185]: this function is a setter not a getter isn't it? @doohanas 🤔

File: components/Slideshow.vue [hunk 5]
```diff
@@ -151,23 +158,60 @@
...
+  })
+
+  const isAutoPlay = computed(() => {
+    return validatedPayload?.value?.images?.length >= 2
+      ? {
+          delay: 5000,
+          disableOnInteraction: false,
+        }
+      : false
+  })
```
Comment (Ibwedagama) [inline components/Slideshow.vue:181]: minor:

this computed doesn't always return Boolean, so `isAutoPlay` may not be the correct name for this data. Should we use `autoPlayConfig` instead? wdyt @doohanas 🤔

**Review by marsellavaleria19 — APPROVED** [review]

File: components/Slideshow.vue (hunk 1 above)
Comment (doohanas) [reply components/Slideshow.vue:117]: Hello @Ibwedagama, yes we already validated, It's fine if we remove optional chaining

File: components/Slideshow.vue (hunk 4 above)
Comment (doohanas) [reply components/Slideshow.vue:185]: oh I See, yes it set value. I change it

File: components/Slideshow.vue (hunk 5 above)
Comment (doohanas) [reply components/Slideshow.vue:181]: Agree it's not always return `boolean` so `isAutoPlay` not exact. We can use `autoPlayConfig`

**Review by yoslie — APPROVED** [review]

File: components/Slideshow.vue (hunk 1 above)
Comment (Ibwedagama) [reply components/Slideshow.vue]: yup I think it's fine @doohanas

**Review by Ibwedagama — APPROVED** [review]

**Pull request merged**